hack/release.sh

```diff
@@ -21,13 +21,9 @@ source $(dirname $0)/../vendor/knative.dev/hack/release.sh
 source $(dirname $0)/build-flags.sh
 
 function build_release() {
-  local ld_flags="$(build_flags $(dirname $0)/..)"
-  local pkg="knative.dev/client/pkg/kn/commands"
-  local version="${TAG}"
-  # Use vYYYYMMDD-<hash>-local for the version string, if not passed.
-  [[ -z "${version}" ]] && version="v${BUILD_TAG}-local"
+  # Env var exported by hack/build-flags.sh
+  local ld_flags="${KN_BUILD_LD_FLAGS:-}"
```

Comment:
An empty default doesn't work well because then you would have a `-ldflags` without value to the option which will very likely lead to an error. IMO you could directly use `KN_BUILD_FLAGS` and commit `ld_flags`. 

Or use something like 

```suggestion
  local ld_flags="-ldflags ${KN_BUILD_LD_FLAGS:-}"
```
 and then just add $ld_flags below, that should also then work with an empty env var.

Reply:
The proposal doesn't work smoothly with our `date` value and quoting. On the other hand I've tested with great success that empty flags from `KN_BUILD_LD_FLAGS` doesn't break the build or produce any error.

Reply:
ok, so let's go with the original approach if this is no problem.

/approve
/lgtm